plugins/pipelines/create.js

```diff
@@ -18,38 +20,51 @@ module.exports = (datastore) => ({
             const scmUrl = request.payload.scmUrl;
             const Pipeline = new Model.Pipeline(datastore);
             const pipelineId = Pipeline.generateId({ scmUrl });
+            const username = request.auth.credentials.username;
+            const User = new Model.User(datastore, password);
 
-            /* eslint-disable consistent-return */
-            Pipeline.get(pipelineId, (error, data) => {
-                if (error) {
-                    return reply(boom.wrap(error));
-                }
-                if (data) {
-                    return reply(boom.conflict('scmUrl needs to be unique'));
-                }
-
-                Pipeline.create(request.payload, (err, result) => {
-                    if (err) {
-                        return reply(boom.wrap(err));
+            async.waterfall([
+                (next) => User.getPermissions({
+                    username,
+                    scmUrl
+                }, next),
+                (permissions, next) => {
+                    if (!permissions.admin) {
+                        return reply(boom.unauthorized(`User ${username}
+                            is not an admin of this repo`));
                     }
 
-                    const location = urlLib.format({
-                        host: request.headers.host,
-                        port: request.headers.port,
-                        protocol: request.server.info.protocol,
-                        pathname: `${request.path}/${result.id}`
-                    });
+                    return Pipeline.get(pipelineId, next);
+                },
+                (data, next) => {
+                    if (data) {
+                        return reply(boom.conflict('scmUrl needs to be unique'));
+                    }
+                    const admins = {};
 
-                    Pipeline.sync({
-                        scmUrl
-                    }, (e) => {
-                        if (e) {
-                            return reply(boom.wrap(e));
-                        }
+                    admins[username] = true;
+                    const pipelineConfig = hoek.merge(request.payload, { admins });
```

Comment:
should we `hoek.applyToDefaults` so we don't merge values into the `request.payload` ?